state-chain/runtime/src/lib.rs

```diff
@@ -2072,8 +2074,48 @@ impl_runtime_apis! {
 		fn cf_validate_refund_params(retry_duration: u32) -> Result<(), DispatchErrorWithMessage> {
 			pallet_cf_swapping::Pallet::<Runtime>::validate_refund_params(retry_duration).map_err(Into::into)
 		}
+
+		fn cf_get_open_deposit_channels(account_id: Option<AccountId>) -> ChainAccounts {
+			let btc_chain_accounts = pallet_cf_ingress_egress::DepositChannelLookup::<Runtime,BitcoinInstance>::iter()
+				.map(|(_, value)| value)
+				.filter(|channel_details| account_id.is_none() || Some(&channel_details.owner) == account_id.as_ref())
+				.filter(|channel_details| match channel_details.action {
+					ChannelAction::Swap {..} => true,
+					ChannelAction::CcmTransfer {..} => true,
+					ChannelAction::LiquidityProvision {..} => false,
```

Comment:
Yes but this rpc is called `cf_get_open_deposit_channels` so it should probably return all of the open deposit channels for the given id. 

If a broker submits this request via the broker api it will implicitly filter out an LP channels anyway. And later on when we want to be able to monitor *all* channels, we won't get any surprises in the behaviour of this rpc.

Reply:
Agree, sounds good.